Sources/EdgeNetworkHandlers/EdgeEventWarning.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import Foundation
+
+// MARK: - EdgeEventWarning
+
+/// Messages from upstreams which could signal that a partial processing of the request occurred
+struct EdgeEventWarning: Codable {
```

Comment:
Based on https://github.com/adobe/aepsdk-edge-ios/pull/110 we would like to expose EdgeEventError to the caller in the completion handler and I think it might become too complicated for the developer to distinguish between the two. Did you also consider keeping them under the same class as until now, so EdgeEventError would have an optional report payload if provided by the server? This way the EdgeEventError is still in line with the AEP Error Response event.

Reply:
I actually believe it would be more misleading/confusing to represent a warning from Konductor with a type that contains the name "Error," indicating it's an error rather than a warning. I do understand the perspective of having more types for the customer to understand could be confusing. However, I find it quite a bit more confusing to have one type that claims it's an error (by name) to actually represent both errors and warnings. We need to indicate what an error is and what is not an error. By separating these, we can clearly label one an error and one a warning. 

If we overload this type to represent both warnings and errors, I think we'll end up with more confusion, and it'll be easy to mistake an error for a warning.

Reply:
So are you suggesting to also have a new event type that an extension should listen to?

Reply:
Yeah, I think that may be a good idea. I believe for total clarity it will be best to be able to easily distinguish between warnings and events.

We can discuss this possibility further and I am happy to submit another PR to add the new event type and change the warning events from errors to warning type.